### PR TITLE
feat: enabled chart tab for button object along with maintaining backward compatibility of properties

### PR DESCRIPTION
**Pull request by rajani-varpe**
This PR intends to enable the chart tab in button object along with backward compatibility of editor properties except the background Image using Media Library.

Background Image using Media Library is working fine in Styling Panel, the action item to get this working with backward compatibility is to add the correct condition in MediaLibrary component. This functionality works fine with "**args.properties**" (MediaLibrary Component)but later this has been changed to "**args.handler.layout.component**" which tend to fail as we do not have any "components" in style editor. 

So this condition needs to be handled to get this working properly.

### 1. Styling Panel

image

### 2. Style Editor - without FF

image

### 3. Styling Panel - with FF

image

### CODE REVIEW

**Comment by Donya-qlik** [comment]
@AmeliaAndersson @LiKang6688 @niekvanstaveren Could you please start reviewing this PR? I'll take care of this PR since the responsible person is not working with Qlik anymore. Thanks!

**Note**: Rajani has mentioned in his explanations that Media Library is not working with `args.layout.properties`. I'm not sure I understood what he meant. Media Library works properly in sense-client as an extension as far as I tested it. Let me know if you have any idea about it. Thanks.

File: src/styling-panel-definition.js [hunk 1]
```diff
@@ -11,7 +44,406 @@ const getStylingPanelDefinition = (bkgOptionsEnabled) => ({
       ref: 'components',
       useGeneral: true,
       useBackground: bkgOptionsEnabled,
+      items: {
+        labelSection: {
+          translation: 'Label',
+          component: 'panel-section',
+          show(data, handler, args) {
+            console.log("show=>labelSection=>", args)
```
Comment (niekvanstaveren) [inline src/styling-panel-definition.js:52]: Remove all console.log statements before merging

File: src/styling-panel-definition.js [hunk 2]
```diff
@@ -11,7 +44,406 @@ const getStylingPanelDefinition = (bkgOptionsEnabled) => ({
       ref: 'components',
       useGeneral: true,
       useBackground: bkgOptionsEnabled,
+      items: {
+        labelSection: {
+          translation: 'Label',
```
Comment (niekvanstaveren) [inline src/styling-panel-definition.js:49]: Is this a real translation?

File: src/styling-panel-definition.js [hunk 3]
```diff
@@ -11,7 +44,406 @@ const getStylingPanelDefinition = (bkgOptionsEnabled) => ({
...
+          component: 'panel-section',
+          show(data, handler, args) {
```
Comment (niekvanstaveren) [inline src/styling-panel-definition.js:51]: What is this show function supposed to do? It just return true in the end, but there is a lot of logic in here which should not be here

File: src/styling-panel-definition.js [hunk 4]
```diff
@@ -11,7 +44,406 @@ const getStylingPanelDefinition = (bkgOptionsEnabled) => ({
...
+              obj.label = args?.handler?.layout?.style?.label;
+              args.properties.components.push(obj)
+            }
+            return true;
+          },
+          items: {
+            labelItem: {
+              component: 'items',
+              ref: 'components',
+              key: 'actionbutton',
```
Comment (niekvanstaveren) [inline src/styling-panel-definition.js:102]: I don't think these need to be in the components array. Only when this is something that could be generic it should be placed in components. The key `actionbutton` already shows that it is probably not generic

File: src/styling-panel-definition.js [hunk 5]
```diff
@@ -11,7 +44,406 @@ const getStylingPanelDefinition = (bkgOptionsEnabled) => ({
...
+                      translation: 'Common.Expression',
+                      show: (data) => propertyResolver.getValue(data, 'font.useColorExpression'),
+                    },
+                  }
+                }
+              },
+            },
+          },
+        },
+        bgColorSection: {
```
Comment (niekvanstaveren) [inline src/styling-panel-definition.js:178]: Why do we have this now both in the general tab and the chart tab?

**Review by niekvanstaveren — CHANGES_REQUESTED** [review]
A couple things here really need to be fixed. I would actually suggest starting a new PR for this.

Have not tested things myself, but I am pretty sure that there will be some issues when using the new options. For instance what would happen if I have a background selected in the old setup and now want to change the font color with the feature flag active. Do I still get the old background settings applied?

**Pull request closed without merging**